### PR TITLE
upgrade Maxima to 5.46.0

### LINKED ISSUE #32898: Maxima SPKG: eliminate matrixexp.patch
<div id="comment:0"></div>

Our custom patch matrixexp.patch
is hindering using Maxima from the system.

Related:

- https://sourceforge.net/p/maxima/bugs/2596/
  --- upstream bug report
- #32867: spkg-configure.m4 for maxima
  --- would benefit from patch removal
- #31553: Maxima SPKG: eliminate
  undoing_true_false_printing_patch.patch
- #13973: Upgrade Maxima to 5.33.0
  --- where the patch was introduced

CC:  @slel @orlitzky

Component: **distribution**

Keywords: **Maxima**

_Issue created by migration from https://trac.sagemath.org/ticket/32898_

### PR DESCRIPTION
**Pull request by dimpase**
### :books: upgrade Maxima to 5.46.0

Our 5.45.0 is getting old. This will fix #32898 as well.

We have one mild regression found while working on #35615 (already reported upstream),
 but number of improvements in a number of places. As #35615 is a part of this PR branch,
I propose to review them all here. 

We also promote `info` spkg to standard, and make it install `makeinfo/texi2any`, to deal with Maxima docs.
This allows us to simplify configuration/installation of Maxima and ECL.

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### :hourglass: Dependencies

#35615 - provide `spkg-configure.m4` for Maxima

### CODE REVIEW

**Comment by dimpase** [comment]
CI fails due to a demand for `makeinfo` (something new for 5.46, I gather):
```
2023-05-06T19:48:36.7969163Z [maxima-5.46.0] checking for ecl... true
2023-05-06T19:48:36.7969636Z [maxima-5.46.0] checking for makeinfo... false
2023-05-06T19:48:36.7970184Z [maxima-5.46.0] configure: error: No makeinfo found; consider --disable-build-docs
```

**Comment by dimpase** [comment]
```
sage -t --random-seed=191226560908718331502074978913271091124 sage/tests/cmdline.py
**********************************************************************
File "sage/tests/cmdline.py", line 576, in sage.tests.cmdline.test_executable
Failed example:
    err
Expected:
    ''
Got:
    ';;; Warning: Maxima is unable to set up the help system.\n(Details: CL-INFO::LOAD-PRIMARY-INDEX: Filesystem error with pathname #P"/sage/local/share/info/maxima-index.lisp".\nEither\n 1) the file does not exist, or\n 2) we are not allowed to access the file, or\n 3) the pathname points to a broken symbolic link.)\n\n'
```

seems that we can't proceed without `makeinfo`...

**Comment by dimpase** [comment]
OK, so we need to decide what to do with failing docs/help thing. The old hack, to avoid calling `makeinfo`, touching this and that, does not work any more (and build breaks due to lack of `makebuild`), maxima's configure option `--disable-build-docs` results in the build proceeding, but not providing the maxima help system.

The cleanest way out for me is to promote `info` spkg to standard, and let it build `makeinfo`, not only `info`.
Then all these hacks are no longer needed, as well as `makeinfo` version check in ECL's `spkg-install.m4` - the latter will be moved to `info`'s `spkg-configure.m4`.

Are you OK with it? If so, I'll go ahead with it. If not, please propose an alternative.

**Comment by dimpase** [comment]
@kiwifb - in particular, your input would be valuable, as you appear to have written the `touch` hacks to avoid building docs..

**Comment by dimpase** [comment]
> Plus, reading from the notes, the makeinfo provided by OS X used to be too old.

it seems that Apple stopped providing `makeinfo` all together. Anyway, while working on an ECL update I wrote [tests](https://github.com/sagemath/sage/blob/8aa721379a5fa8f7daff9fa8c636d10dbedb2fce/build/pkgs/ecl/spkg-install.in#L35) to check for `makeinfo` to be new enough, and we can reuse them.

**Comment by dimpase** [comment]
> unwillingness from anyone to require and then have makeinfo in the sage tree. If we bit that bullet, it should be fine.

need to build (make)info would be limited to purists willing to build everything in Sage from scratch. Surely makeinfo and info is available in any sane building environment.

**Comment by kiwifb** [comment]
I vaguely remember working with David Kirkby on this. His solaris environment was not really standard. And because the object were already present, the other unwillingness was to waste the time rebuilding them. In any case, we have moved to a point where asking for makeinfo is not onerous. Let us move on.

**Comment by dimpase** [comment]
> unwillingness from anyone to require and then have makeinfo in the sage tree. If we bit that bullet, it should be fine.

The update is being tested now. The need to build (make)info would be limited to purists willing to build everything in Sage from scratch. Surely `makeinfo` and `info` are available in any sane building environment.

**Comment by mkoeppe** [comment]
+1 on making the `info` spkg standard and building makeinfo.

File: build/pkgs/info/spkg-configure.m4 [hunk 1]
```diff
@@ -1,4 +1,14 @@
 SAGE_SPKG_CONFIGURE([info], [
   AC_PATH_PROG(INFO, info)
```
Comment (mkoeppe) [inline build/pkgs/info/spkg-configure.m4:2]: perhaps add a test for the makeinfo program?

Comment (dimpase) [reply build/pkgs/info/spkg-configure.m4:2]: It is there, just look further down the patch.

**Comment by dimpase** [comment]
OK, so this test run only returns the test errors coming from 5.46 being different from 5.45. Will be fixed later today.
{{{
sage -t --random-seed=312649298022371725183055163141926694063 sage/calculus/calculus.py  # 5 doctests failed
sage -t --random-seed=312649298022371725183055163141926694063 sage/functions/exp_integral.py  # 1 doctest failed
sage -t --random-seed=312649298022371725183055163141926694063 sage/interfaces/maxima_lib.py  # 2 doctests failed
sage -t --random-seed=312649298022371725183055163141926694063 sage/symbolic/integration/integral.py  # 1 doctest failed
sage -t --random-seed=312649298022371725183055163141926694063 sage/symbolic/relation.py  # 1 doctest failed
}}}

File: build/pkgs/info/spkg-configure.m4 (hunk 1 above)
Comment (mkoeppe) [reply build/pkgs/info/spkg-configure.m4:2]: Indeed, sorry, I shouldn't comment before finishing my morning coffee

**Comment by kiwifb** [comment]
The test failures are all in #33718 along with various discussion including link to upstream bug report. It is interesting reading and not without reminding of a recent thread about treatment of number on sage-devel.

**Comment by orlitzky** [comment]
Two of the failures on #33718 look like bad tests to begin with. For example instead of testing that the string representation of `2*y*arctan(1.0/y) - 2*y*arctan(0.0001414/y) + 1.0*log(1.0*y^2 + 1.0) - 0.0001414*log(1.0*y^2 + 1.9993959999999997e-08) - 1.9997172` never changes, we could instead put the "expected" answer in `expected`, the actual one in `actual`, and then check that `(expected - actual)(y=some number)` is zeroish.

**Comment by dimpase** [comment]
OK, all ready now. To accommodate both 5.45 and 5.46, a couple of tests had to be made a little weaker, because
we don't have a proper way to version tests, depending on a component version.

**Comment by dimpase** [comment]
@isuruf Conda runs here fail due to a conda package conflict involving different OpenSSL versions. Just in case - you probably saw this already.

**Comment by mkoeppe** [comment]
@dimpase I think the dependencies are backwards. PR #35615 by itself cannot be merged because tests will fail with system maxima 5.46.0

**Comment by mkoeppe** [comment]
Also maxima spkg-configure.m4 needs to do some version check, as I said in https://github.com/sagemath/sage/pull/35615#issuecomment-1537190576

File: src/sage/functions/exp_integral.py [hunk 2]
```diff
@@ -601,8 +601,8 @@ class Function_log_integral_offset(BuiltinFunction):
         1/log(x)
         sage: f.integrate(x)
         -x*log_integral(2) + x*log_integral(x) - Ei(2*log(x))
-        sage: Li(x).integrate(x,2.0,4.5)
-        -2.5*log_integral(2) + 5.799321147411334
+        sage: Li(x).integrate(x,2.0,4.5).n()
+        3.18641169711760
```
Comment (orlitzky) [inline src/sage/functions/exp_integral.py]: Add an `abs tol` here as a preventative measure? The following test uses `1e-15`.

File: build/pkgs/maxima/spkg-configure.m4 [hunk 3]
```diff
@@ -1,8 +1,21 @@
 SAGE_SPKG_CONFIGURE([maxima], [
+  m4_pushdef([SAGE_MAXIMA_MINVER],["5.45.0"])dnl this version and higher allowed
   SAGE_SPKG_DEPCHECK([ecl], [
     dnl First check for the "maxima" executable in the user's PATH, because
     dnl we still use pexpect to communicate with it in a few places.
-    AC_PATH_PROG([SAGE_MAXIMA], [maxima])
+    AC_CACHE_CHECK([for Maxima >= $SAGE_MAXIMA_MINVER], [ac_cv_path_MAXIMA], [
+        AC_PATH_PROGS_FEATURE_CHECK([MAXIMA], [maxima], [
+            maxima_version=`$ac_path_MAXIMA --version 2>&1 | tail -1\
```
Comment (orlitzky) [inline build/pkgs/maxima/spkg-configure.m4]: `tail -1` only works with GNU tail but `tail -n 1` should be portable

File: src/sage/functions/exp_integral.py (hunk 2 above)
Comment (dimpase) [reply src/sage/functions/exp_integral.py]: It's actually an exact value, `-Ei(2*log(9/2)) + Ei(2*log(2)) + 9/2*log_integral(9/2) - 9/2*log_integral(2)` - as computed by maxima 5.46.  I gather that the discrepancy is due to 5.45 and 5.46 treating floating point input a bit differently.

I can cut few digits and replace them with `...`

File: build/pkgs/maxima/spkg-configure.m4 (hunk 3 above)
Comment (dimpase) [reply build/pkgs/maxima/spkg-configure.m4]: fixed.

**Comment by dimpase** [comment]
> Could we maybe split the "fix doctests" part and try to merge it for 10.0 as we've done before?
> 
we're very close to 10.0, I doubt Volker would be willing to go for it. Chances that something won't work on an obscure platform are >0. @vbraun ?

> The rest (maxima update / accept maxima from system in sage-the-distro) can be pushed soon after 10.0 is released.
> 
> Indeed sagelib is already working fine with maxima from system in many distros; however we are either stuck with 5.45.1 or with failing doctests.

File: src/sage/calculus/calculus.py [hunk 4]
```diff
@@ -136,10 +136,11 @@
     [          1/2*(e^(2*sqrt(x)) + 1)*e^(x - sqrt(x)) 1/2*(x*e^(2*sqrt(x)) - x)*sqrt(x)*e^(x - sqrt(x))]
     [  1/2*(e^(2*sqrt(x)) - 1)*e^(x - sqrt(x))/x^(3/2)           1/2*(e^(2*sqrt(x)) + 1)*e^(x - sqrt(x))]
 
-And complex exponentiation works now::
+Complex exponentiation works, but may require a patched version of
+maxima (:trac:`32898`) for now::
 
     sage: M = i*matrix([[pi]])
-    sage: e^M
+    sage: e^M # not tested, requires patched maxima
```
Comment (tornaria) [inline src/sage/calculus/calculus.py]:
```suggestion
    sage: e^M  # not tested, requires patched maxima
```
two spaces ?

File: src/sage/calculus/calculus.py [hunk 5]
```diff
@@ -1203,7 +1214,7 @@ def limit(ex, dir=None, taylor=False, algorithm='maxima', **argv):
 
         sage: var('a')
         a
-        sage: limit(x^a,x=0)
+        sage: limit(x^a,x=0) # random - maxima 5.46.0 does not need extra assumption
```
Comment (tornaria) [inline src/sage/calculus/calculus.py]:
```suggestion
        sage: limit(x^a,x=0)  # random - maxima 5.46.0 does not need extra assumption
```

File: src/sage/calculus/calculus.py [hunk 6]
```diff
@@ -1215,7 +1226,7 @@ def limit(ex, dir=None, taylor=False, algorithm='maxima', **argv):
     With this example, Maxima is looking for a LOT of information::
 
         sage: assume(a>0)
-        sage: limit(x^a,x=0)
+        sage: limit(x^a,x=0) # random - maxima 5.46.0 does not need extra assumption
```
Comment (tornaria) [inline src/sage/calculus/calculus.py]:
```suggestion
        sage: limit(x^a,x=0)  # random - maxima 5.46.0 does not need extra assumption
```

File: src/sage/calculus/calculus.py [hunk 7]
```diff
@@ -1224,7 +1235,7 @@ def limit(ex, dir=None, taylor=False, algorithm='maxima', **argv):
          more details)
         Is a an integer?
         sage: assume(a,'integer')
-        sage: limit(x^a, x=0)
+        sage: limit(x^a, x=0) # random - maxima 5.46.0 does not need extra assumption
```
Comment (tornaria) [inline src/sage/calculus/calculus.py]:
```suggestion
        sage: limit(x^a, x=0)  # random - maxima 5.46.0 does not need extra assumption
```

File: src/sage/interfaces/maxima_lib.py [hunk 8]
```diff
@@ -947,7 +954,7 @@ def sr_limit(self, expr, v, a, dir=None):
             for more details)
             Is a positive, negative or zero?
             sage: assume(a>0)
-            sage: limit(x^a,x=0)
+            sage: limit(x^a,x=0) # random - not needed for maxima 5.46.0
```
Comment (tornaria) [inline src/sage/interfaces/maxima_lib.py]:
```suggestion
            sage: limit(x^a,x=0)  # random - not needed for maxima 5.46.0
```

File: src/sage/matrix/matrix2.pyx [hunk 9]
```diff
@@ -15425,9 +15425,10 @@ cdef class Matrix(Matrix1):
 
         TESTS:
 
-        Check that sparse matrices are handled correctly (:trac:`28935`)::
+        Sparse matrices are handled correctly (:trac:`28935`), but may
+        require a patched version of maxima (:trac:`32898`) for now::
 
-            sage: matrix.diagonal([0], sparse=True).exp()
+            sage: matrix.diagonal([0], sparse=True).exp() # not tested, requires patched maxima
```
Comment (tornaria) [inline src/sage/matrix/matrix2.pyx]:
```suggestion
            sage: matrix.diagonal([0], sparse=True).exp()  # not tested, requires patched maxima
```

File: src/sage/interfaces/maxima.py [hunk 10]
```diff
@@ -622,21 +622,13 @@ def _start(self):
...
-            sage: b = a^600              # long time -- about 10-15 seconds
-
         """
         Expect._start(self)
         self._sendline(r":lisp (defun tex-derivative (x l r) (tex (if $derivabbrev (tex-dabbrev x) (tex-d x '\\partial)) l r lop rop ))")
 
         # Don't use ! for factorials (#11539)
         self._sendline(":lisp (remprop 'mfactorial 'grind)")
 
-        # Remove limit on the max heapsize (since otherwise it defaults
```
Comment (mkoeppe) [inline src/sage/interfaces/maxima.py:637]: is this no longer needed?

Comment (dimpase) [reply src/sage/interfaces/maxima.py:637]: no. cf. the commit message: https://github.com/sagemath/sage/commit/45c13ed8ebcd7e1f21977d0b176eba40e063fd4a

File: src/sage/functions/exp_integral.py (hunk 2 above)
Comment (kiwifb) [reply src/sage/functions/exp_integral.py]: I am more in favor of adding a tolerance myself. Numerical results like this one should always have a tolerance. As an alternative, we could use, say, `.n(digits=10)` which is not unlike `...` but more purposeful in my opinion.

Comment (dimpase) [reply src/sage/functions/exp_integral.py]: it's not really a numerical result, in the sense it's evaluating a symbolic expression in functions Maxima knows how to evaluate well.
I'll add `digits=10`.

Comment (dimpase) [reply src/sage/functions/exp_integral.py]: fixed by [79d7c28](https://github.com/sagemath/sage/pull/35619/commits/79d7c2862d0841d50ded76a47ead039d59a321e1)

**Review by kiwifb — APPROVED** [review]
I personally approve of the change on sagelib side. They work for me and I'll ship them in sage-on-gentoo for sage-10.0.
The changes on the distribution part for maxima and its dependencies look appropriate but I did not test them in real life.

**Comment by orlitzky** [comment]
Does this really work without assumptions? (What answer does it return?)

```
sage: a = var('a')
sage: limit(x^a,x=0)  # random - maxima 5.46.0 does not need extra assumption
```

The latest sage beta with maxima-5.46.0 on Gentoo still demands assumptions from me. And the answer *does* depend on the sign of `a`:

```
sage: assume(a > 0)
sage: limit(x^a,x=0)
0
sage: forget()
sage: assume(a <= 0, a >= 0)
sage: limit(x^a,x=0)
1
```

**Comment by dimpase** [comment]
if I remove `# random` tags, as follows:
```
--- a/src/sage/calculus/calculus.py
+++ b/src/sage/calculus/calculus.py
@@ -1214,7 +1214,7 @@ def limit(ex, dir=None, taylor=False, algorithm='maxima', **argv):
 
         sage: var('a')
         a
-        sage: limit(x^a,x=0) # random - maxima 5.46.0 does not need extra assumption
+        sage: limit(x^a,x=0)
         Traceback (most recent call last):
         ...
         ValueError: Computation failed since Maxima requested additional
@@ -1226,7 +1226,7 @@ def limit(ex, dir=None, taylor=False, algorithm='maxima', **argv):
     With this example, Maxima is looking for a LOT of information::
 
         sage: assume(a>0)
-        sage: limit(x^a,x=0) # random - maxima 5.46.0 does not need extra assumption
+        sage: limit(x^a,x=0)
         Traceback (most recent call last):
         ...
         ValueError: Computation failed since Maxima requested additional
@@ -1235,7 +1235,7 @@ def limit(ex, dir=None, taylor=False, algorithm='maxima', **argv):
          more details)
         Is a an integer?
         sage: assume(a,'integer')
-        sage: limit(x^a, x=0) # random - maxima 5.46.0 does not need extra assumption
+        sage: limit(x^a, x=0)
         Traceback (most recent call last):
         ...
         ValueError: Computation failed since Maxima requested additional
```
then I get (with Maxima 5.46.0 from Gentoo, by the way, just like you have):

```
sage -t --warn-long 64.8 --random-seed=249651691122430839423715329500238939924 src/sage/calculus/calculus.py
**********************************************************************
File "src/sage/calculus/calculus.py", line 1229, in sage.calculus.calculus.limit
Failed example:
    limit(x^a,x=0)
Expected:
    Traceback (most recent call last):
    ...
    ValueError: Computation failed since Maxima requested additional
    constraints; using the 'assume' command before evaluation *may* help
    (example of legal syntax is 'assume(a>0)', see `assume?` for
     more details)
    Is a an integer?
Got:
    0
**********************************************************************
File "src/sage/calculus/calculus.py", line 1238, in sage.calculus.calculus.limit
Failed example:
    limit(x^a, x=0)
Expected:
    Traceback (most recent call last):
    ...
    ValueError: Computation failed since Maxima requested additional
    constraints; using the 'assume' command before evaluation *may* help
    (example of legal syntax is 'assume(a>0)', see `assume?` for
     more details)
    Is a an even number?
Got:
    0
**********************************************************************
1 item had failures:
   2 of  68 in sage.calculus.calculus.limit
    [458 tests, 2 failures, 14.09 s]
----------------------------------------------------------------------
sage -t --warn-long 64.8 --random-seed=249651691122430839423715329500238939924 src/sage/calculus/calculus.py  # 2 doctests failed
```

I must say I didn't try the interactive computation, so this looks like an artefact of doctesting...

**Comment by collares** [comment]
I think @orlitzky was just talking about the first `#random` tag (the one where no assumptions are made on `a`) , not all three of them!

File: src/sage/calculus/calculus.py [hunk 11]
```diff
@@ -1203,7 +1214,7 @@ def limit(ex, dir=None, taylor=False, algorithm='maxima', **argv):
 
         sage: var('a')
         a
-        sage: limit(x^a,x=0)
+        sage: limit(x^a,x=0)  # random - maxima 5.46.0 does not need extra assumption
```
Comment (collares) [inline src/sage/calculus/calculus.py]:
```suggestion
        sage: limit(x^a,x=0)
```
I think orlitzky was talking about this particular `# random` tag. The following ones, which assume `a` is positive, indeed changed between 5.45 and 5.46 (tested on NixOS).

Comment (dimpase) [reply src/sage/calculus/calculus.py]: oops, right, I miscounted the number of assumptions it still needs. Good catch.

**Comment by orlitzky** [comment]
Ok that makes more sense, thanks. With that out of the way, I think we should drop the remaining `# random` limit tests:

1. The example in `calculus.py` already serves its purpose, to demonstrate that "Maxima may ask for more information," without the `# random` follow-ups.
2. The `# random` test in `maxima_lib.py` is under `TESTS::`, so it isn't shown to users, and the output isn't tested -- it's just wasting time.

**Comment by dimpase** [comment]
I actually don't understand why there are so many  tests in `maxima_lib.py` repeated in `calculus.py`.
Is it because long time ago one was via `pexpect`, and the other not?

**Comment by dimpase** [comment]
@nbruin - could you comment on massive test duplication in `maxima_lib.py` and `calculus.py` ? Can we drop some?

**Comment by nbruin** [comment]
> @nbruin - could you comment on massive test duplication in `maxima_lib.py` and `calculus.py` ? Can we drop some?

I think most test coverage in maxima_lib is specific to testing the "binary" interface and its peculiarities and otherwise mostly to satisfy coverage.

Are you thinking about sr_integrate and sr_sum etc (stubs to facilitate maxima_lib interfacing from calculus.py)? Those may have accumulated some tests, but mainly from people inserting tests for bugfixes. I would be surprised if people would have inserted those in two spots, so I would not expect them to duplicate much.

The documentation of maxima_lib was developed with maxima_lib, and I don't think much from calculus.py was duplicated into it. There may be some stuff in maxima_abstract which probably inherited more legacy from when calculus was using expect interface.

**Comment by dimpase** [comment]
No, I'm thinking about e.g. the doctests  `limit(x^a,x=0)` - which is the same codepath in `calculus.py` and in `maxima_lib.py`, right?

**Comment by nbruin** [comment]
> No, I'm thinking about e.g. the doctests `limit(x^a,x=0)` - which is the same codepath in `calculus.py` and in `maxima_lib.py`, right?

Isn't that the only substantial test that is replicated between the two files? As far as I can see it was introduced in https://github.com/sagemath/sage/issues/7377 in https://github.com/sagemath/sage-prod/files/10646689/trac_7377-assumptions-p3.patch.gz . You can probably find in the long discussion on that ticket what the rationale was for copying the test.

I agree that identical tests in different files are still going to test the same code path (in both cases the code gets run in the standard doctesting scope).

**Comment by mkoeppe** [comment]
Build fails on `fedora-32-standard` - https://github.com/mkoeppe/sage/actions/runs/5036258960/jobs/9032290531#step:14:9916
```
Making all in doc
Making all in info
./build_html.sh
./build_html.sh: abs_srcdir=/sage/local/var/tmp/sage/build/maxima-5.46.0/src/doc/info, *** language=en, languagedir=/sage/local/var/tmp/sage/build/maxima-5.46.0/src/doc/info, texiinitfile=/sage/local/var/tmp/sage/build/maxima-5.46.0/src/doc/info/texi2html.init
./build_html.sh
./build_html.sh: abs_srcdir=/sage/local/var/tmp/sage/build/maxima-5.46.0/src/doc/info, *** language=en, languagedir=/sage/local/var/tmp/sage/build/maxima-5.46.0/src/doc/info, texiinitfile=/sage/local/var/tmp/sage/build/maxima-5.46.0/src/doc/info/texi2html.init
./build_html.sh: line 91: $(basename $includedfile_absolute): No such file or directory
./build_html.sh: line 91: $(basename $includedfile_absolute): No such file or directory
./build_html.sh: line 91: $(basename $includedfile_absolute): No such file or directory
```
To inspect: `docker run -it ghcr.io/mkoeppe/sage/sage-fedora-32-standard-with-targets-pre:8a8e6ce782-failed bash`

**Comment by dimpase** [comment]
Fedora 32 had been EOL'ed for about a year when Maxima 5.46.0 came out.

**Comment by dimpase** [comment]
also, texinfo 6.8 came out after Fedora 32 was EOL'ed. 

Fixing potential issues with software released after certain OS was EOLed is beyond the scope of this PR, and should not be a priority of the project in the 1st place.

**Comment by mkoeppe** [comment]
As I explained to you before, old Fedora serves as a proxy for RHEL.

**Comment by mkoeppe** [comment]
> texinfo 6.8

What does this version refer to?

File: build/pkgs/tox/spkg-configure.m4 [hunk 12]
```diff
@@ -5,7 +5,7 @@ SAGE_SPKG_CONFIGURE([tox], [
        m4_pushdef([TOX4_MIN_VERSION], [4.0.15])
        AC_CACHE_CHECK([for tox 3 >= ]TOX3_MIN_VERSION[ or tox 4 >= ]TOX4_MIN_VERSION, [ac_cv_path_TOX], [
          AC_PATH_PROGS_FEATURE_CHECK([TOX], [tox], [
-            tox_version=$($ac_path_TOX --version 2> /dev/null | tail -1)
+            tox_version=$($ac_path_TOX --version 2> /dev/null | tail -n -1)
```
Comment (mkoeppe) [inline build/pkgs/tox/spkg-configure.m4]: Should this not be `-n 1`?

File: build/pkgs/maxima/spkg-configure.m4 [hunk 13]
```diff
@@ -0,0 +1,46 @@
+SAGE_SPKG_CONFIGURE([maxima], [
+  m4_pushdef([SAGE_MAXIMA_MINVER],["5.45.0"])dnl this version and higher allowed
+  SAGE_SPKG_DEPCHECK([ecl], [
+    dnl First check for the "maxima" executable in the user's PATH, because
+    dnl we still use pexpect to communicate with it in a few places.
+    AC_CACHE_CHECK([for Maxima >= $SAGE_MAXIMA_MINVER], [ac_cv_path_MAXIMA], [
+        AC_PATH_PROGS_FEATURE_CHECK([MAXIMA], [maxima], [
+            maxima_version=`$ac_path_MAXIMA --version 2>&1 | tail -n -1\
+                | $SED -n -e 's/Maxima *\([[0-9]]*\.[[0-9]]*\.[[0-9]]*\)/\1/p'`
```
Comment (mkoeppe) [inline build/pkgs/maxima/spkg-configure.m4:9]: This should be `tail -n 1`, I think

**Comment by dimpase** [comment]
texinfo 6.8 is the Sage's version. Should update to 7.0 by the way

**Comment by mkoeppe** [comment]
Thanks. So fedora-32 installs an older version, which our configure script rejects correctly because it does not have `texi2any`.

**Comment by mkoeppe** [comment]
In a second run on fedora-32, https://github.com/mkoeppe/sage/actions/runs/5039255345/jobs/9037285001, it succeeds.
So this may be a Makefile parallelization bug in Maxima.

**Comment by mkoeppe** [comment]
I see the following on `fedora-39` (https://github.com/mkoeppe/sage/actions/runs/5039255345/jobs/9037285404#step:13:10763)
```
aking all in doc
Making all in info
./build_html.sh
./build_html.sh: abs_srcdir=/sage/local/var/tmp/sage/build/maxima-5.46.0/src/doc/info, *** language=en, languagedir=/sage/local/var/tmp/sage/build/maxima-5.46.0/src/doc/info, texiinitfile=/sage/local/var/tmp/sage/build/maxima-5.46.0/src/doc/info/texi2html.init
./build_html.sh: execute category program: "ecl" --shell make-categories.lisp
makeinfo: warning: set_from_cmdline: unknown variable INLINE_CONTENTS
Category-Help.texi:33: @ref reference to nonexistent node `Item: Introduction/deffn/quad_control'
./build_html.sh
./build_html.sh: abs_srcdir=/sage/local/var/tmp/sage/build/maxima-5.46.0/src/doc/info, *** language=en, languagedir=/sage/local/var/tmp/sage/build/maxima-5.46.0/src/doc/info, texiinitfile=/sage/local/var/tmp/sage/build/maxima-5.46.0/src/doc/info/texi2html.init
makeinfo: warning: set_from_cmdline: unknown variable INLINE_CONTENTS
could not open maxima.texi: No such file or directory
rm: cannot remove 'tmp_html': Directory not empty
make[6]: *** [Makefile:1638: maxima_toc.html] Error 1
```

**Comment by mkoeppe** [comment]
So probably we should build maxima with `make -j1`

File: build/pkgs/maxima/spkg-configure.m4 (hunk 13 above)
Comment (dimpase) [reply build/pkgs/maxima/spkg-configure.m4:9]: fixed

**Comment by mkoeppe** [comment]
> if ecl comes from the system, then info might not be there yet.

No, that's not what happened: https://github.com/mkoeppe/sage/actions/runs/5039255345/jobs/9037285404#step:10:6296
```
[info-6.8] installing. Log file: /sage/logs/pkgs/info-6.8.log
  [gap-4.12.2] successfully installed.
make --no-print-directory jmol-SAGE_LOCAL-no-deps
[jmol-14.29.52] installing. Log file: /sage/logs/pkgs/jmol-14.29.52.log
  [jmol-14.29.52] successfully installed.
make --no-print-directory libbraiding-SAGE_LOCAL-no-deps
[libbraiding-1.1] installing. Log file: /sage/logs/pkgs/libbraiding-1.1.log
  [libbraiding-1.1] successfully installed.
make --no-print-directory linbox-SAGE_LOCAL-no-deps
[linbox-1.6.3.p1] installing. Log file: /sage/logs/pkgs/linbox-1.6.3.p1.log
  [info-6.8] successfully installed.
...
[maxima-5.46.0] installing. Log file: /sage/logs/pkgs/maxima-5.46.0.log
```

**Comment by mkoeppe** [comment]
Thanks. Build is looking good now on all Linux platforms - https://github.com/mkoeppe/sage/actions/runs/5051686843

File: build/pkgs/giac/spkg-configure.m4 [hunk 14]
```diff
@@ -5,7 +5,7 @@ SAGE_SPKG_CONFIGURE([giac], [
        m4_pushdef([GIAC_MAX_VERSION], [1.9.999])
        AC_CACHE_CHECK([for giac >= ]GIAC_MIN_VERSION[, <= ]GIAC_MAX_VERSION, [ac_cv_path_GIAC], [
          AC_PATH_PROGS_FEATURE_CHECK([GIAC], [giac], [
-            giac_version=$($ac_path_GIAC --version 2> /dev/null | tail -1)
+            giac_version=$($ac_path_GIAC --version 2> /dev/null | tail -n -1)
```
Comment (mkoeppe) [inline build/pkgs/giac/spkg-configure.m4]: Here's another `-n -1` that should be `-n 1`

Comment (dimpase) [reply build/pkgs/giac/spkg-configure.m4]: ok

**Review by mkoeppe — APPROVED** [review]
LGTM re portability.
I haven't followed the discussion regarding the tests.

**Comment by orlitzky** [comment]
> I haven't followed the discussion regarding the tests.

They're not a deal-breaker; it's fine by me as well.

A few of the `EXAMPLES` that fail with maxima-5.46 now have `# random` tags, but they don't show anything new, and will be bad examples in the future because you can only follow along with an old maxima. Similarly I think we had one `TESTS` which now has a `# random` tag. But an invisible test that isn't tested is just wasting cycles.

**Comment by mkoeppe** [comment]
Then it sounds like we can set it to positive review. I'll do this now, but if there other concerns, as usual please feel free to revert.

**Comment by mkoeppe** [comment]
> So probably we should build maxima with `make -j1`

@dimpase, we need to do this fix as a follow-up. This build error has showed up again in the 10.1.beta1 test - 
https://github.com/sagemath/sage/actions/runs/5103657597/jobs/9174057935#step:13:9737

**Comment by dimpase** [comment]
is it this: 
https://sourceforge.net/p/maxima/bugs/3699/
?

a ecl bug/feature?

**Comment by mkoeppe** [comment]
I think it's simply that https://sourceforge.net/p/maxima/code/ci/master/tree/doc/info/Makefile.am executes two copies of `./build-html.sh` via the rule in line 696. These two copies then can race in creating and removing the directory `tmp_html`

**Comment by dimpase** [comment]
if so, it's easy to fix, just split this into two rules, and make the 2nd of them depend on the 1st, no?

**Comment by orlitzky** [comment]
...do we need to build the docs? I think they're all still in the release tarball, despite what `./configure` implies. The `./configure` error stems from,

```
commit 74168af0d0b3f3be0f2fd0f46f638e45a84a4f42
Author: Robert Dodier <robert_dodier@users.sourceforge.net>
Date:   Sun Nov 21 05:57:35 2021 -0800

    Implement configure option to omit building documentation.

     * configure.ac: recognize new option --enable-build-docs (default = yes)
           and define ENABLE_BUILD_DOCS
     * Makefile.am: execute make in top-level doc directory only if ENABLE_BUILD_DOCS
     * interfaces/xmaxima/Makefile.am: execute make in doc direcory only if ENABLE_BUILD_DOCS
```

which can be disabled with `--disable-build-docs`.

**Comment by dimpase** [comment]
I'd rather build the docs, it's quick, and more flexible. 

We specifically promoted info spkg to standard, and made it install/check for makeinfo.

**Comment by orlitzky** [comment]
Ok, I see now that you already had `--disable-build-docs` in an earlier commit and removed it.

**Comment by mkoeppe** [comment]
> if so, it's easy to fix, just split this into two rules, and make the 2nd of them depend on the 1st, no?

Sure. Do you want to work on it?

**Comment by dimpase** [comment]
will do

**Comment by mkoeppe** [comment]
I've put a fix on #35652

**Comment by mkoeppe** [comment]
Also, the new dependency on `info` for `ecl` and `maxima` should have been made an order-only dependency. 

Not very important, but I'll fix this in #35652, where the incremental workflow is unnecessarily building `ecl`

**Pull request merged**